packages/utils/telemetry-utils/src/config.ts

```diff
@@ -199,18 +200,22 @@ export class CachedConfigProvider implements IConfigProvider {
     }
 
     private getCacheEntry(name: string): StronglyTypedValue | undefined {
-        if (!this.configCache.has(name)) {
+        // Generate version specific name by appending package version. Version specific setting always take
+        // precedence over general setting for a feature gate.
+        const versionSpecificConfigName = `${name}_${pkgVersion}`;
```

Comment:
I was thinking about this, I think the version should come from caller no? because that version could be different from the version of this package? @vladsud @andre4i

Reply:
That's a great insight, Jatin! However, our packages are in sync with `telemetry-utils` version-wise, right? Ideally, I agree that the caller would provide the version but as long as the system is in sync, we shouldn't have a problem. I'm also not very excited about the API changes required to enable providing the version..

I do see an issue for layer-spanning feature gates, but I'm not sure if we have an instance of it right now. Say the runtime and the loader are using different versions but they're reading the same feature gate. If the feature gate is supposed to consistently change the behavior in both layers, this scenario will break.

Reply:
started a similar conversation in the other PR: https://github.com/microsoft/FluidFramework/pull/12240#discussion_r991501668

Reply:
this version will always be the loader, not the driver or runtime, as the loaded is what builds our config provider

Reply:
We either need to keep the api same, and the caller would just append its package version in the name or we can change the api and make the version argument optional. I think second approach should be fine as the caller is then not forced to supply the version and then the common settings would be provided and then it won't be a breaking change either.

Reply:
Brining my comments over from the other PR: 

> i've been thinking about something similar, but i don't this this is the correct approach. this package version here will not be consistent across the layers, as this will be loader, not driver or runtime. we also have an issue that any change in version, patch, update, etc will essentially disable all existing configurations. 

>I think this problem needs to be tackled more wholistically in terms about how we are thinking about versions across our layers and data a rest. I hope we can find a workable scheme that is simple, like using integer generations which are easy to define as base lines. 

> for instance, in the case of configurations we'd express the config value in a way that expresses for all generations after x, apply this config, maybe like `<generation>.My.Config`. 

> Maybe that will be too simple, and we need something more expressive like a semantic version, and sem ver (~,^). But regardless of the solution, this seems like more general problem we need to invest in.

Reply:
+ @scarlettjlee who is looking at some other things related to versioning, and may have insights or find this conversation useful